pkg/client/interfaces.go

```diff
@@ -230,9 +230,10 @@ func (sc *StatusConfig) Validate() error {
 
 // PreflightConfig are the options passed to PreflightChecks.
 type PreflightConfig struct {
-	Namespace    string
-	DNSNamespace string
-	DNSPodLabels []string
+	Namespace       string
+	DNSNamespace    string
+	DNSPodLabels    []string
+	PreflightChecks map[string]func(kubernetes.Interface, *PreflightConfig) error
```

Comment:
I think it may be easier to push the processing of the `skip` logic into `preflight.go`.

So instead of getting the map ahead of time, why not just put the list of preflight skips into this struct (rather than the map).

Then your caller only needs to pass along that string slice (easy) and then the sonobuoy client, when launching the checks can do the bulk of the logic.

I like the way you're doing it though; getting a copy of the valid func map, then deleting the values listed in the slice.